### PR TITLE
Test and lint opentelemetry-resource-detector-azure

### PR DESCRIPTION
**Pull request by xrmx**
# Description

Seems a good idea to run tests and lint on opentelemetry-resource-detector-azure

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A

# Does This PR Require a Core Repo Change?

- [ ] Yes. - Link to PR: 
- [x] No.

# Checklist:

See [contributing.md](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/CONTRIBUTING.md) for styleguide, changelog guidelines, and more.

- [x] Followed the style guidelines of this project
- [ ] Changelogs have been updated
- [ ] Unit tests have been added
- [ ] Documentation has been updated

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -11,6 +11,11 @@ envlist =
     pypy3-test-resource-detector-container
     lint-resource-detector-container
 
+    ; opentelemetry-resource-detector-azure
+    py3{8,9,10,11,12}-test-resource-detector-azure
+    pypy3-test-resource-detector-azure
+    lint-resource-detector-azure
```
Comment (emdneto) [inline tox.ini:17]: I suppose we should add the steps to lint as well right?

**Review by lzchen — APPROVED** [review]

**Review by ocelotl — APPROVED** [review]

File: tox.ini (hunk 1 above)
Comment (xrmx) [reply tox.ini:17]: Ah, do you mean to add the commands right?

**Review by emdneto — APPROVED** [review]

File: tox.ini [hunk 2]
```diff
@@ -744,6 +749,12 @@ commands_pre =
   resource-detector-container: pip install opentelemetry-test-utils@{env:CORE_REPO}\#egg=opentelemetry-test-utils&subdirectory=tests/opentelemetry-test-utils
   resource-detector-container: pip install -r {toxinidir}/resource/opentelemetry-resource-detector-container/test-requirements.txt
 
+  resource-detector-azure: pip install opentelemetry-api@{env:CORE_REPO}\#egg=opentelemetry-api&subdirectory=opentelemetry-api
+  resource-detector-azure: pip install opentelemetry-semantic-conventions@{env:CORE_REPO}\#egg=opentelemetry-semantic-conventions&subdirectory=opentelemetry-semantic-conventions
+  resource-detector-azure: pip install opentelemetry-sdk@{env:CORE_REPO}\#egg=opentelemetry-sdk&subdirectory=opentelemetry-sdk
+  resource-detector-azure: pip install opentelemetry-test-utils@{env:CORE_REPO}\#egg=opentelemetry-test-utils&subdirectory=tests/opentelemetry-test-utils
+  lint-resource-detector-azure: pip install opentelemetry-instrumentation@{env:CORE_REPO}\#egg=opentelemetry-instrumentation&subdirectory=opentelemetry-instrumentation
```
Comment (emdneto) [inline tox.ini:756]: I think we are missing install the package itself for tests works

Comment (xrmx) [reply tox.ini]: That has been fixed in the last push

**Pull request merged**